counterexamples/sorgenfrey_line.lean

```diff
@@ -0,0 +1,290 @@
+  simp only [continuous_at, nhds_prod_eq, nhds_eq_map, nhds_eq_comap (x + y), prod_map_map_eq,
+    tendsto_comap_iff, tendsto_map'_iff, (∘), ← nhds_within_prod_eq],
+  exact (continuous_add.tendsto _).inf (maps_to.tendsto $ λ x hx, add_le_add hx.1 hx.2)
+end
+
+lemma is_clopen_Ici (a : ℝₗ) : is_clopen (Ici a) := ⟨is_open_Ici a, is_closed_Ici⟩
+
+lemma is_clopen_Iio (a : ℝₗ) : is_clopen (Iio a) :=
+by simpa only [compl_Ici] using (is_clopen_Ici a).compl
+
+lemma is_clopen_Ico (a b : ℝₗ) : is_clopen (Ico a b) :=
+(is_clopen_Ici a).inter (is_clopen_Iio b)
+
+instance : totally_disconnected_space ℝₗ :=
+⟨λ s hs' hs x hx y hy, le_antisymm (hs.subset_clopen (is_clopen_Ici x) ⟨x, hx, le_rfl⟩ hy)
+  (hs.subset_clopen (is_clopen_Ici y) ⟨y, hy, le_rfl⟩ hx)⟩
+
+instance : first_countable_topology ℝₗ := ⟨λ x, (nhds_basis_Ico_rat x).is_countably_generated⟩
+
+/-- Sorgenfrey line is a normal topological space. -/
+instance : normal_space ℝₗ :=
+begin
+  /- Let `s` and `t` be disjoint closed sets. For each `x ∈ s` we choose `X x` such that
+  `set.Ico x (X x)` is disjoint with `t`. Similarly, for each `y ∈ t` we choose `Y y` such that
+  `set.Ico y (Y y)` is disjoint with `s`. Then `⋃ x ∈ s, Ico x (X x)` and `⋃ y ∈ t, Ico y (Y y)` are
+  disjoint open sets that include `s` and `t`. -/
+  refine ⟨λ s t hs ht hd, _⟩,
+  choose! X hX hXd using λ x (hx : x ∈ s), exists_Ico_disjoint_closed ht (disjoint_left.1 hd hx),
+  choose! Y hY hYd using λ y (hy : y ∈ t), exists_Ico_disjoint_closed hs (disjoint_right.1 hd hy),
+  refine ⟨⋃ x ∈ s, Ico x (X x), ⋃ y ∈ t, Ico y (Y y), is_open_bUnion $ λ _ _, is_open_Ico _ _,
+    is_open_bUnion $ λ _ _, is_open_Ico _ _, _, _, _⟩,
+  { exact λ x hx, mem_Union₂.2 ⟨x, hx, left_mem_Ico.2 $ hX x hx⟩ },
+  { exact λ y hy, mem_Union₂.2 ⟨y, hy, left_mem_Ico.2 $ hY y hy⟩ },
+  { simp only [disjoint_Union_left, disjoint_Union_right, Ico_disjoint_Ico],
+    intros y hy x hx,
+    clear hs ht hd hX hY,
+    cases le_total x y with hle hle,
+    { calc min (X x) (Y y) ≤ X x : min_le_left _ _
+      ... ≤ y : not_lt.1 (λ hyx, hXd x hx ⟨⟨hle, hyx⟩, hy⟩)
+      ... ≤ max x y : le_max_right _ _ },
+    { calc min (X x) (Y y) ≤ Y y : min_le_right _ _
+      ... ≤ x : not_lt.1 $ λ hxy, hYd y hy ⟨⟨hle, hxy⟩, hx⟩
+      ... ≤ max x y : le_max_left _ _ } }
+end
+
+lemma dense_range_coe_rat : dense_range (coe : ℚ → ℝₗ) :=
+begin
+  refine dense_iff_inter_open.2 _,
+  rintro U Uo ⟨x, hx⟩,
+  rcases is_open_iff.1 Uo _ hx with ⟨y, hxy, hU⟩,
+  rcases exists_rat_btwn hxy with ⟨z, hxz, hzy⟩,
+  exact ⟨z, hU ⟨hxz.le, hzy⟩, mem_range_self _⟩
+end
+
+instance : separable_space ℝₗ := ⟨⟨_, countable_range _, dense_range_coe_rat⟩⟩
+
+lemma is_closed_antidiagonal (c : ℝₗ) : is_closed {x : ℝₗ × ℝₗ | x.1 + x.2 = c} :=
+is_closed_singleton.preimage continuous_add
+
+lemma is_clopen_Ici_prod (x : ℝₗ × ℝₗ) : is_clopen (Ici x) :=
+(Ici_prod_eq x).symm ▸ (is_clopen_Ici _).prod (is_clopen_Ici _)
+
+lemma is_closed_of_subset_antidiagonal {s : set (ℝₗ × ℝₗ)} {c : ℝₗ}
```

Comment:
can you add a docstring to this lemma, because it is both important and striking?